src/autogluon/cloud/backend/sagemaker_backend.py

```diff
@@ -1104,15 +1112,6 @@ def _predict_real_time(self, test_data, accept, split_pred_proba=True, inference
             raise e
 
     def _upload_batch_predict_data(self, test_data, bucket, key_prefix):
-        # If a directory of images, upload directly
-        if isinstance(test_data, str) and not os.path.isdir(test_data):
-            # either a file to a dataframe, or a file to an image
-            if is_image_file(test_data):
-                logger.warning(
-                    "Are you sure you want to do batch inference on a single image? You might want to try `deploy()` and `predict_real_time()` instead"
-                )
-            else:
-                test_data = load_pd.load(test_data)
```

Comment:
Will it cause problem if images are uploaded here?

Reply:
These logics have been copied over to line #L1184 - #L1192

Reply:
I figured. Now after I read it again, I think moving this to `_predict()` makes sense.